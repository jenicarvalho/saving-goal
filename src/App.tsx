import React from 'react';
import Header from './components/Header';
import SavingGoal from './pages/SavingGoal';

import { globalStyle as GlobalStyle } from './styles/global';

export function App(): JSX.Element {
  return (
    <div>
      <Header />
      <SavingGoal />
      <GlobalStyle />
    </div>
  );
}
