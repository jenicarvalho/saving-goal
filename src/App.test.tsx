import { App } from './App';
import { render } from '@testing-library/react';

describe('App', () => {
  describe('using testing library', () => {
    it('returns the text', () => {
      const component = render(<App />);

      expect(component).toMatchSnapshot();
    });
  });
});
