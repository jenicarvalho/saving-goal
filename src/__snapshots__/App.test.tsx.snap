// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App using testing library returns the text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-bdnxRM iXQQfK"
        >
          <svg
            fill="none"
            viewBox="0 0 100 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M47.434 27.9331L50.6378 24.8262C54.4942 28.5193 60.7238 28.5193 64.5506 24.8262L67.7544 27.9331C62.1774 33.3556 53.0406 33.3556 47.434 27.9331ZM86.3245 6.59492L85.6719 4.48454H81.8155V22.5106H86.8288V14.0398C86.8288 9.26219 90.003 8.32425 91.7829 8.41218C95.4316 8.58805 94.957 13.1898 94.957 13.8054V22.4813H100V12.7502C100 7.70872 97.6268 4.22075 92.7321 4.22075C89.5283 4.19144 87.0958 5.68628 86.3245 6.59492ZM43.1029 22.5106V4.48454H38.2676V22.5106H43.1029ZM76.9801 22.5106V4.48454H72.1448V22.5106H76.9801ZM27.7662 6.38974L27.084 4.48454H23.2275V22.5106H28.2705V13.395C28.2705 10.0243 31.0294 8.7346 33.6992 8.7346C34.3518 8.7346 34.9748 8.82253 35.3604 8.88115V4.19144H34.1738C31.059 4.19144 28.5969 5.59835 27.7662 6.38974ZM19.3414 13.4829C19.3414 7.73804 14.8621 4.22075 9.75972 4.22075C4.68704 4.19144 0 7.70872 0 13.5416C0 19.2865 4.47938 22.8038 9.61139 22.8038C14.7434 22.8038 19.3414 19.2865 19.3414 13.4829ZM4.95402 13.5122C4.95402 10.6691 6.88223 8.44149 9.75972 8.44149C12.6372 8.44149 14.3874 10.6691 14.3874 13.5122C14.3874 16.3554 12.6372 18.583 9.75972 18.583C6.88223 18.583 4.95402 16.3554 4.95402 13.5122ZM65.3812 7.73804C66.5678 9.23288 67.2797 11.1674 67.2797 13.4536C67.2797 19.2572 62.652 22.8038 57.5497 22.8038C52.388 22.8038 47.9383 19.2572 47.9383 13.5416C47.9383 12.9847 47.968 12.4571 48.057 11.9002C48.2349 10.8743 48.5613 9.90703 49.0359 8.9984C49.5105 8.14839 50.1038 7.38631 50.8454 6.74147C51.5574 6.09663 52.3583 5.56904 53.2483 5.15869C54.1382 4.74834 55.0875 4.45523 56.0664 4.30868C57.253 4.13281 58.5583 4.19144 59.8339 4.19144C60.3678 4.19144 60.8425 4.22075 61.3764 4.13281C62.1774 4.01557 62.8597 3.66384 63.4233 3.10694C64.2539 2.28624 64.6692 1.14312 64.6692 0H69.2969C69.2969 3.25349 67.7544 6.12594 65.3812 7.73804ZM62.3257 13.5122C62.3257 10.6691 60.5755 8.44149 57.698 8.44149C54.8205 8.44149 52.8923 10.6691 52.8923 13.5122C52.8923 16.3554 54.8205 18.583 57.698 18.583C60.5755 18.583 62.3257 16.3554 62.3257 13.5122Z"
              fill="#1E2A32"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="sc-iCoGMd bzbUhZ"
        >
          <h1
            class="sc-fujyAs cUEzvX"
          >
            Lets plan your 
            <strong>
              saving goal
            </strong>
          </h1>
          <div
            class="sc-jrsJWt expVOV"
          >
            <div
              class="sc-gtsrHT fVOJFn"
            >
              <img
                alt="House goal"
                src="buy-a-house.svg"
              />
              <div>
                <h2>
                  Buy a house
                </h2>
                <span>
                  Saving goal
                </span>
              </div>
            </div>
            <div
              class="sc-dlnjwi ianbHZ"
            >
              <div
                class="sc-hKFxyN zFDVE"
              >
                <label>
                  Total amount
                </label>
                <div
                  class="sc-kEqXSa fDUGYW"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 0C12.5523 0 13 0.447715 13 1V23C13 23.5523 12.5523 24 12 24C11.4477 24 11 23.5523 11 23V1C11 0.447715 11.4477 0 12 0Z"
                      fill="#CBD5DC"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M6.31802 5.31802C7.16193 4.47411 8.30653 4 9.5 4H17C17.5523 4 18 4.44772 18 5C18 5.55228 17.5523 6 17 6H9.5C8.83696 6 8.20107 6.26339 7.73223 6.73223C7.26339 7.20107 7 7.83696 7 8.5C7 9.16304 7.26339 9.79893 7.73223 10.2678C8.20107 10.7366 8.83696 11 9.5 11H14.5C15.6935 11 16.8381 11.4741 17.682 12.318C18.5259 13.1619 19 14.3065 19 15.5C19 16.6935 18.5259 17.8381 17.682 18.682C16.8381 19.5259 15.6935 20 14.5 20H6C5.44772 20 5 19.5523 5 19C5 18.4477 5.44772 18 6 18H14.5C15.163 18 15.7989 17.7366 16.2678 17.2678C16.7366 16.7989 17 16.163 17 15.5C17 14.837 16.7366 14.2011 16.2678 13.7322C15.7989 13.2634 15.163 13 14.5 13H9.5C8.30653 13 7.16193 12.5259 6.31802 11.682C5.47411 10.8381 5 9.69347 5 8.5C5 7.30653 5.47411 6.16193 6.31802 5.31802Z"
                      fill="#CBD5DC"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <input
                    data-testid="input-amount"
                    name="amount"
                    type="text"
                    value="0.00"
                  />
                </div>
              </div>
              <div
                class="sc-hKFxyN zFDVE"
              >
                <label>
                  Reach goal by
                </label>
                <div
                  class="sc-kEqXSa fDUGYW"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        class=""
                        name="reachDate"
                        type="text"
                        value="February"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sc-eCApnc gsjGYK"
            >
              <div
                class="sc-jSFjdj gAhFOM"
              >
                <h3>
                  Monthly amount
                </h3>
                <span
                  id="result-formated"
                >
                  $0.00
                </span>
              </div>
              <div
                class="sc-gKAaRy gTOKlm"
              >
                Youâ€™re planning 
                <strong>
                  2
                   monthly deposits
                </strong>
                 to reach your 
                <strong>
                  $0.00
                </strong>
                 goal by
                 
                <strong>
                  February 2022
                </strong>
                .
              </div>
            </div>
            <div
              class="sc-pNWdM cXBkSg"
            >
              <button>
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-bdnxRM iXQQfK"
      >
        <svg
          fill="none"
          viewBox="0 0 100 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M47.434 27.9331L50.6378 24.8262C54.4942 28.5193 60.7238 28.5193 64.5506 24.8262L67.7544 27.9331C62.1774 33.3556 53.0406 33.3556 47.434 27.9331ZM86.3245 6.59492L85.6719 4.48454H81.8155V22.5106H86.8288V14.0398C86.8288 9.26219 90.003 8.32425 91.7829 8.41218C95.4316 8.58805 94.957 13.1898 94.957 13.8054V22.4813H100V12.7502C100 7.70872 97.6268 4.22075 92.7321 4.22075C89.5283 4.19144 87.0958 5.68628 86.3245 6.59492ZM43.1029 22.5106V4.48454H38.2676V22.5106H43.1029ZM76.9801 22.5106V4.48454H72.1448V22.5106H76.9801ZM27.7662 6.38974L27.084 4.48454H23.2275V22.5106H28.2705V13.395C28.2705 10.0243 31.0294 8.7346 33.6992 8.7346C34.3518 8.7346 34.9748 8.82253 35.3604 8.88115V4.19144H34.1738C31.059 4.19144 28.5969 5.59835 27.7662 6.38974ZM19.3414 13.4829C19.3414 7.73804 14.8621 4.22075 9.75972 4.22075C4.68704 4.19144 0 7.70872 0 13.5416C0 19.2865 4.47938 22.8038 9.61139 22.8038C14.7434 22.8038 19.3414 19.2865 19.3414 13.4829ZM4.95402 13.5122C4.95402 10.6691 6.88223 8.44149 9.75972 8.44149C12.6372 8.44149 14.3874 10.6691 14.3874 13.5122C14.3874 16.3554 12.6372 18.583 9.75972 18.583C6.88223 18.583 4.95402 16.3554 4.95402 13.5122ZM65.3812 7.73804C66.5678 9.23288 67.2797 11.1674 67.2797 13.4536C67.2797 19.2572 62.652 22.8038 57.5497 22.8038C52.388 22.8038 47.9383 19.2572 47.9383 13.5416C47.9383 12.9847 47.968 12.4571 48.057 11.9002C48.2349 10.8743 48.5613 9.90703 49.0359 8.9984C49.5105 8.14839 50.1038 7.38631 50.8454 6.74147C51.5574 6.09663 52.3583 5.56904 53.2483 5.15869C54.1382 4.74834 55.0875 4.45523 56.0664 4.30868C57.253 4.13281 58.5583 4.19144 59.8339 4.19144C60.3678 4.19144 60.8425 4.22075 61.3764 4.13281C62.1774 4.01557 62.8597 3.66384 63.4233 3.10694C64.2539 2.28624 64.6692 1.14312 64.6692 0H69.2969C69.2969 3.25349 67.7544 6.12594 65.3812 7.73804ZM62.3257 13.5122C62.3257 10.6691 60.5755 8.44149 57.698 8.44149C54.8205 8.44149 52.8923 10.6691 52.8923 13.5122C52.8923 16.3554 54.8205 18.583 57.698 18.583C60.5755 18.583 62.3257 16.3554 62.3257 13.5122Z"
            fill="#1E2A32"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <div
        class="sc-iCoGMd bzbUhZ"
      >
        <h1
          class="sc-fujyAs cUEzvX"
        >
          Lets plan your 
          <strong>
            saving goal
          </strong>
        </h1>
        <div
          class="sc-jrsJWt expVOV"
        >
          <div
            class="sc-gtsrHT fVOJFn"
          >
            <img
              alt="House goal"
              src="buy-a-house.svg"
            />
            <div>
              <h2>
                Buy a house
              </h2>
              <span>
                Saving goal
              </span>
            </div>
          </div>
          <div
            class="sc-dlnjwi ianbHZ"
          >
            <div
              class="sc-hKFxyN zFDVE"
            >
              <label>
                Total amount
              </label>
              <div
                class="sc-kEqXSa fDUGYW"
              >
                <svg
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 0C12.5523 0 13 0.447715 13 1V23C13 23.5523 12.5523 24 12 24C11.4477 24 11 23.5523 11 23V1C11 0.447715 11.4477 0 12 0Z"
                    fill="#CBD5DC"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M6.31802 5.31802C7.16193 4.47411 8.30653 4 9.5 4H17C17.5523 4 18 4.44772 18 5C18 5.55228 17.5523 6 17 6H9.5C8.83696 6 8.20107 6.26339 7.73223 6.73223C7.26339 7.20107 7 7.83696 7 8.5C7 9.16304 7.26339 9.79893 7.73223 10.2678C8.20107 10.7366 8.83696 11 9.5 11H14.5C15.6935 11 16.8381 11.4741 17.682 12.318C18.5259 13.1619 19 14.3065 19 15.5C19 16.6935 18.5259 17.8381 17.682 18.682C16.8381 19.5259 15.6935 20 14.5 20H6C5.44772 20 5 19.5523 5 19C5 18.4477 5.44772 18 6 18H14.5C15.163 18 15.7989 17.7366 16.2678 17.2678C16.7366 16.7989 17 16.163 17 15.5C17 14.837 16.7366 14.2011 16.2678 13.7322C15.7989 13.2634 15.163 13 14.5 13H9.5C8.30653 13 7.16193 12.5259 6.31802 11.682C5.47411 10.8381 5 9.69347 5 8.5C5 7.30653 5.47411 6.16193 6.31802 5.31802Z"
                    fill="#CBD5DC"
                    fill-rule="evenodd"
                  />
                </svg>
                <input
                  data-testid="input-amount"
                  name="amount"
                  type="text"
                  value="0.00"
                />
              </div>
            </div>
            <div
              class="sc-hKFxyN zFDVE"
            >
              <label>
                Reach goal by
              </label>
              <div
                class="sc-kEqXSa fDUGYW"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class=""
                      name="reachDate"
                      type="text"
                      value="February"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-eCApnc gsjGYK"
          >
            <div
              class="sc-jSFjdj gAhFOM"
            >
              <h3>
                Monthly amount
              </h3>
              <span
                id="result-formated"
              >
                $0.00
              </span>
            </div>
            <div
              class="sc-gKAaRy gTOKlm"
            >
              Youâ€™re planning 
              <strong>
                2
                 monthly deposits
              </strong>
               to reach your 
              <strong>
                $0.00
              </strong>
               goal by
               
              <strong>
                February 2022
              </strong>
              .
            </div>
          </div>
          <div
            class="sc-pNWdM cXBkSg"
          >
            <button>
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
